### ISSUE TITLE
First few requests never get the authorization header

### ISSUE THREAD

**Issue by artlowel** [DSpace/dspace-angular#756]
When you are authenticated, and you refresh, the first few rest requests won't contain the authorization header. 

The problem is easiest to see in dev mode, as you can simply inspect the requests in the dev tools and look at their headers, but it also happens in production mode, on the express server.

If you add the following statement to [auth.interceptor.ts, line 251](https://github.com/DSpace/dspace-angular/blob/b731301b39f0ebf9c8c2a6401702033ccbe0ab31/src/app/core/auth/auth.interceptor.ts#L251) you can see in the server output that it happens during server side rendering as well
```typescript
console.log(newReq.url, 'authorized', isNotEmpty(newReq.headers.get('authorization')));
```

This is problematic because as we're adding support for things like the authorization endpoint to decide which menu options to show etc, the responses to those very first requests depend on having the correct authentication info.

**Comment by atarix83** [comment]
@artlowel A possible solution to this problem, should be to active [router-outlet](https://github.com/DSpace/dspace-angular/blob/b731301b39f0ebf9c8c2a6401702033ccbe0ab31/src/app/app.component.html#L20) only once the [auth.loaded state](https://github.com/DSpace/dspace-angular/blob/b731301b39f0ebf9c8c2a6401702033ccbe0ab31/src/app/core/auth/auth.reducer.ts#L37) is true. In this way it should prevent to load any route, thereby to make new request, before an attempt to restore an existing auth session is done.

**Comment by artlowel** [comment]
@atarix83 explained in the meeting today that we could also wrap a container with an ngIf="auth.loaded" around everything in app component instead of just the router-outlet. That should prevent any other requests from being fired too soon

**Comment by artlowel** [comment]
assigned myself, as we'll fix this in the context of #635
